docs/source/asr/datasets.rst

```diff
@@ -507,6 +507,127 @@ An example using an AIS cluster at ``hostname:port`` with a tarred dataset for t
+* Dynamic weighted data source multiplexing
+    An approach to combining diverse data sources (e.g. multiple domains, languages, tasks)
+    where each data source is treated as a separate stream with its own sampling probability. The resulting data stream is a
+    multiplexer that samples from each sub-stream. This approach ensures that the distribution of different sources is approximately
+    constant in time (i.e., stationary); in fact, each mini-batch will have roughly the same ratio of data coming from each source.
+    Since the multiplexing is done dynamically, it is very easy to tune the sampling weights.
+
+Lhotse dataloading supports the following types of inputs:
+
+* NeMo manifests
+    Regular NeMo JSON manifests.
+* NeMo tarred data
+    Tarred NeMo JSON manifests + audio tar files; we also support combination of multiple NeMo
+    tarred data sources (e.g., multiple buckets of NeMo data or multiple datasets) via dynamic multiplexing.
+* Lhotse CutSet manifests
+    Regular Lhotse CutSet manifests (typically gzipped JSONL).
+    See `Lhotse Cuts documentation`_ to learn more about Lhotse data formats.
+* Lhotse Shar data
+    Lhotse Shar is a data format that also uses tar files for sequential data loading,
+    but is designed to be modular (i.e., easily extensible with new data sources and with new feature fields).
+    More details can be found here: |tutorial_shar|
+
+.. caution:: As of now, Lhotse is mainly supported in most ASR model configurations. We aim to gradually extend this support to other speech tasks.
+
+.. _Lhotse: https://github.com/lhotse-speech/lhotse
+.. _Lhotse Cuts documentation: https://lhotse.readthedocs.io/en/latest/cuts.html
+.. |tutorial_shar| image:: https://colab.research.google.com/assets/colab-badge.svg
+    :target: https://colab.research.google.com/github/lhotse-speech/lhotse/blob/master/examples/04-lhotse-shar.ipynb
+
+Enabling Lhotse via configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Start with an existing NeMo experiment YAML configuration. Typically, you'll only need to add a few options to enable Lhotse.
+These options are::
+
+    # NeMo generic dataloading arguments
+    model.train_ds.manifest_filepath=...
+    model.train_ds.tarred_audio_filepaths=...   # for tarred datasets only
+    model.train_ds.num_workers=4
+    model.train_ds.min_duration=0.3             # optional
+    model.train_ds.max_duration=30.0            # optional
+    model.train_ds.shuffle=true                 # optional
+
+    # Lhotse dataloading related arguments
+    ++model.train_ds.use_lhotse=True
+    ++model.train_ds.batch_duration=1100
+    ++model.train_ds.quadratic_duration=30
+    ++model.train_ds.num_buckets=30
+    ++model.train_ds.num_cuts_for_bins_estimate=10000
+    ++model.train_ds.bucket_buffer_size=10000
+    ++model.train_ds.shuffle_buffer_size=10000
+
+    # PyTorch Lightning related arguments
+    ++trainer.use_distributed_sampler=false
```

Comment:
Can we set this config automatically whenever we detect use_lhotse=true?

Reply:
I wanted to do that but it's tricky because it's affecting the trainer config. `Trainer` is the first object instantiated in the training scripts so we'd need to modify the start of every training script with something like `if cfg...use_lhotse: trainer.use_distributed_sampler=False`, e.g. here:

https://github.com/NVIDIA/NeMo/blob/6082d760b9d66a787d8328d88de6c301fe49a7e2/examples/asr/speech_to_text_finetune.py#L188-L192

Changing the setting after `Trainer` is created won't have an effect because `Trainer` would have already created another object in its `__init__` that depends on this option..

https://github.com/Lightning-AI/pytorch-lightning/blob/75e112f138ec5cdd8eee2c26720b480719f415d6/src/lightning/pytorch/trainer/trainer.py#L400-L411